src/aics-image-viewer/components/App/index.tsx

```diff
@@ -788,100 +799,69 @@ export default class App extends React.Component<AppProps, AppState> {
     });
   }
 
-  // TODO make nicer and more strictly type-able
-  handleChangeToImage(keyToChange: string, newValue: any, index?: number) {
-    const { image, userSelections, view3d } = this.state;
-    if (!image || !view3d) {
+  private userSelectionChangeHandlers: UserSelectionChangeHandlers = {
+    mode: (mode, view3d) => view3d.setCameraMode(mode),
+    maxProject: (value, view3d, image) => {
+      view3d.setMaxProjectMode(image, value);
+      view3d.updateActiveChannels(image);
+    },
+    pathTrace: (enabled, view3d, image) => {
+      view3d.setVolumeRenderMode(enabled ? RENDERMODE_PATHTRACE : RENDERMODE_RAYMARCH);
+      view3d.updateActiveChannels(image);
+    },
+
+    showAxes: (showing, view3d) => view3d.setShowAxis(showing),
+    showBoundingBox: (showing, view3d, image) => view3d.setShowBoundingBox(image, showing),
+    boundingBoxColor: (color, view3d, image) => view3d.setBoundingBoxColor(image, colorArrayToFloats(color)),
+    backgroundColor: (color, view3d) => view3d.setBackgroundColor(colorArrayToFloats(color)),
+
+    alphaMaskSliderLevel: (value, view3d, image) => {
+      view3d.updateMaskAlpha(image, alphaSliderToImageValue(value));
+      view3d.updateActiveChannels(image);
+    },
+    brightnessSliderLevel: (value, view3d) => {
```

Comment:
should typescript warn when there's no third parameter?

Reply:
Callbacks passed to `map` or `reduce` nominally take three or four arguments, but the last one is rarely defined. Given that example alone I think it's reasonable for TypeScript to accept functions that define fewer arguments than they'll actually be called with. In this specific case though, I'm with you in being somewhat unsure of this choice stylistically; I'd absolutely be open to explicitly filling in those unused parameters, e.g. `brightnessSliderLevel: (value, view3d, _image) => { ... }`

Reply:
I think just adding the underscore param would make it more obvious that we expect all the function signatures to be the same